apps/pie-docs/src/_11ty/shortcodes/card.js

```diff
@@ -1,45 +1,99 @@
 const pieIconsSvg = require('../filters/pieIconsSvg');
+const pieDesignTokenColours = require('../filters/pieDesignTokenColours');
 
 /**
  * A Card HTML component – takes an array of list items and turns them into a list of cards
  * @param {string[]} items - An array of card items
  * @returns {string}
  */
 // eslint-disable-next-line func-names
-module.exports = function ({ items }) {
-    const iconLink = pieIconsSvg({
-        name: 'link-external',
-        attrs: {
-            'aria-hidden': 'true',
-            height: 21,
-            width: 21,
-        },
-    });
+module.exports = function ({ items, shouldFillContainer = false }) {
+    const buildLinkIcon = (isInternalLink) => {
+        const internalLinkIcon = pieIconsSvg({
+            name: 'arrow-right',
+            attrs: {
+                'aria-hidden': 'true',
+                height: 16,
+                width: 16,
+            },
+        });
+
+        const externalLinkIcon = pieIconsSvg({
+            name: 'link-external',
+            attrs: {
+                'aria-hidden': 'true',
+                height: 21,
+                width: 21,
+            },
+        });
+
+        return isInternalLink ? internalLinkIcon : externalLinkIcon;
+    };
+
+    const buildCardContent = ({
+        icon, iconColour, heading, content,
+    }) => {
+        const iconStyles = iconColour ? `style="--icon-container-colour: ${pieDesignTokenColours({ tokenName: iconColour, tokenPath: ['alias', 'default'] })};"` : '';
+
+        const cardIcon = icon && pieIconsSvg({
+            name: icon,
+            attrs: {
+                'aria-hidden': 'true',
+                height: 32,
+                width: 32,
+            },
+        });
+
+        return `<div class="c-card-container">
+                    ${cardIcon ? `<div class="c-card-icon" ${iconStyles}>${cardIcon}</div>` : ''}
+                    ${heading ? `<h2 class="c-card-heading">${heading}</h2>` : ''}
```

Comment:
perhaps we should default to an h2, but let the call provide a heading level. Then this could be `<h${headingLevel}>`. This might make it a bit more page structure-friendly